nat-lab/tests/test_direct_connection.py

```diff
@@ -39,265 +40,227 @@
-    pytest.param(
-        _generate_setup_parameter_pair((a[0], a[1], batch_a), (b[0], b[1], batch_b)),
-    )
-    for (a, b) in UHP_FAILING_PATHS_PARAMS
-    for (batch_a, batch_b) in itertools.product(DISABLED_BATCHING_OPTIONS, repeat=2)
+# fmt: off
+UHP_WORKING_PATHS_PARAMS = [
+    [
+        (ConnectionTag.DOCKER_FULLCONE_CLIENT_1, ["stun"], True, DOCKER_FULLCONE_GW_1_IP),
+        (ConnectionTag.DOCKER_FULLCONE_CLIENT_2, ["stun"], False, DOCKER_FULLCONE_GW_2_IP),
+        (ConnectionTag.DOCKER_CONE_CLIENT_1, ["stun"], True, DOCKER_CONE_GW_1_IP),
+        (ConnectionTag.DOCKER_CONE_CLIENT_2, ["stun"], False, DOCKER_CONE_GW_2_IP),
+        (ConnectionTag.DOCKER_OPEN_INTERNET_CLIENT_1, ["stun"], True, DOCKER_OPEN_INTERNET_CLIENT_1_IP),
+        (ConnectionTag.DOCKER_OPEN_INTERNET_CLIENT_2, ["stun"], False, DOCKER_OPEN_INTERNET_CLIENT_2_IP),
+        (ConnectionTag.DOCKER_OPEN_INTERNET_CLIENT_DUAL_STACK, ["stun"], False, DOCKER_OPEN_INTERNET_CLIENT_DUAL_STACK_IP),
+        (ConnectionTag.DOCKER_UPNP_CLIENT_1, ["upnp"], True, DOCKER_UPNP_GW_1_IP),
+        (ConnectionTag.DOCKER_UPNP_CLIENT_2, ["upnp"], False, DOCKER_UPNP_GW_2_IP),
+    ],
+    [
+        (ConnectionTag.DOCKER_SYMMETRIC_CLIENT_1, ["stun"], True, DOCKER_SYMMETRIC_GW_1_IP),
+        (ConnectionTag.DOCKER_FULLCONE_CLIENT_1, ["stun"], True, DOCKER_FULLCONE_GW_1_IP),
+        (ConnectionTag.DOCKER_FULLCONE_CLIENT_2, ["stun"], False, DOCKER_FULLCONE_GW_2_IP),
+        (ConnectionTag.DOCKER_OPEN_INTERNET_CLIENT_1, ["stun"], True, DOCKER_OPEN_INTERNET_CLIENT_1_IP),
+        (ConnectionTag.DOCKER_OPEN_INTERNET_CLIENT_2, ["stun"], False, DOCKER_OPEN_INTERNET_CLIENT_2_IP),
+        (ConnectionTag.DOCKER_OPEN_INTERNET_CLIENT_DUAL_STACK, ["stun"], True, DOCKER_OPEN_INTERNET_CLIENT_DUAL_STACK_IP),
+    ],
+    [
+        (ConnectionTag.DOCKER_OPEN_INTERNET_CLIENT_1, ["local"], True, DOCKER_OPEN_INTERNET_CLIENT_1_IP),
+        (ConnectionTag.DOCKER_SYMMETRIC_CLIENT_1, ["local"], True, DOCKER_OPEN_INTERNET_CLIENT_1_IP),
+    ],
+    [
+        (ConnectionTag.DOCKER_SYMMETRIC_CLIENT_1, ["local"], True, DOCKER_SYMMETRIC_CLIENT_1_IP),
+        (ConnectionTag.DOCKER_INTERNAL_SYMMETRIC_CLIENT, ["local"], True, DOCKER_SYMMETRIC_CLIENT_1_IP),
+    ],
 ]
+# fmt: on
 
 
-@pytest.mark.asyncio
-@pytest.mark.long
-@pytest.mark.timeout(timeouts.TEST_DIRECT_FAILING_PATHS_TIMEOUT)
-@pytest.mark.parametrize("setup_params", UHP_FAILING_PATHS)
-# Not sure this is needed. It will only be helpful to catch if any
-# libtelio change would make any of these setup work.
-async def test_direct_failing_paths(setup_params: List[SetupParameters]) -> None:
-    async with AsyncExitStack() as exit_stack:
-        env = await setup_mesh_nodes(exit_stack, setup_params, is_timeout_expected=True)
-        _, beta = env.nodes
-        alpha_client, beta_client = env.clients
-        alpha_connection, _ = [conn.connection for conn in env.connections]
+async def _ping_between_all_nodes(env: Environment) -> None:
```

Comment:
might be nice to move this to `helpers.py`, so we could use this in other tests, kinda nice util